src/data/LedgerCache.cpp

```diff
@@ -139,6 +176,29 @@ LedgerCache::get(ripple::uint256 const& key, uint32_t seq) const
     return {e->second.blob};
 }
 
+std::optional<Blob>
+LedgerCache::getDeleted(ripple::uint256 const& key, uint32_t seq) const
+{
+    if (disabled_)
+        return std::nullopt;
+
+    std::shared_lock const lck{mtx_};
```

Comment:
Could it be that this method is called so often so it blocks updating the cache?

Reply:
Should not be the case. This is only called for deleted objects on each ledger update 👍